public/js/components/notification.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { Component, PropTypes } from 'react';
+import { gettext } from 'utils';
+
+import cx from 'classnames';
+
+
+export default class Notification extends Component {
+
+  static propTypes = {
+    autoHide: PropTypes.bool,
+    handleAutoHide: React.PropTypes.func,
+    handleDismissClick: React.PropTypes.func,
```

Comment:
In looking at this more, I don't think you need to bind it either. You could just call `this.props.removeNotification(this.props.key)`.